src/Icons/Models/IconHttpResponse.cs

```diff
@@ -0,0 +1,81 @@
+﻿#nullable enable
+
+using System.Collections;
+using System.Net;
+using AngleSharp.Html.Parser;
+using Bit.Icons.Services;
+
+namespace Bit.Icons.Models;
+
+public class IconHttpResponse : IEnumerable<Icon>, IDisposable
+{
+    private const int _maxIconLinksProcessed = 200;
+    private const int _maxRetrievedIcons = 10;
+
+    private readonly HttpResponseMessage _response;
+    private readonly ILogger<IIconFetchingService> _logger;
+    private readonly IHttpClientFactory _httpClientFactory;
+    private readonly List<Icon> _icons = new();
+
+    public HttpStatusCode StatusCode => _response.StatusCode;
+    public bool IsSuccessStatusCode => _response.IsSuccessStatusCode;
+    public string? ContentType => _response.Content.Headers.ContentType?.MediaType;
+    public HttpContent Content => _response.Content;
+    public Icon this[int i]
+    {
+        get
+        {
+            return _icons[i];
+        }
+    }
+    public IEnumerator<Icon> GetEnumerator() => ((IEnumerable<Icon>)_icons).GetEnumerator();
+    IEnumerator IEnumerable.GetEnumerator() => ((IEnumerable)_icons).GetEnumerator();
+
+    public IconHttpResponse(HttpResponseMessage response, ILogger<IIconFetchingService> logger, IHttpClientFactory httpClientFactory)
+    {
+        _response = response;
+        _logger = logger;
+        _httpClientFactory = httpClientFactory;
+    }
+
+    public async Task<IEnumerable<Icon>> RetrieveIconsAsync(Uri requestUri, string domain)
```

Comment:
💭 Does it make sense to just have this return one? Is there a use case I'm not seeing for providing all of what's found?

Reply:
hmm this is pretty interesting. We used to prioritize icon display after pulling, but now that prioritization occurs prior to navigating to a link. That does mean that we could pull only the highest priority and cut down on network requests.

However, if the first link fails, we'd have to go on down the line iteratively. Currently, we just fire off all 10 requests simultaneously and wait for them all to get back to us. Effectively we're discussing changing this from an eager load to a lazy one.

I don't have the data to say which would be better, presumably most links work, but those that don't would suffer in load times.